crates/events/src/enclave_event/compute_request/trbfv.rs

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+//
+// This file is provided WITHOUT ANY WARRANTY;
+// without even the implied warranty of MERCHANTABILITY
+// or FITNESS FOR A PARTICULAR PURPOSE.
+
+use serde::{Deserialize, Serialize};
+
+/// TrBFV modules defining the API for multithreaded compute
+/// Each module defines the event payloads that make up a compute request
+/// Each compute request should live independently and be self contained
+
+/// Input format for TrBFVRequest
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub enum TrBFVRequest {
+    GenEsiSss(gen_esi_sss::Request),
+    GenPkShareAndSkSss(gen_pk_share_and_sk_sss::Request),
+    GenDecryptionKey(gen_decryption_key::Request),
+    GenDecryptionShare(gen_decryption_share::Request),
+    ThresholdDecrypt(threshold_decrypt::Request),
+}
+
+/// Result format for TrBFVResponse
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub enum TrBFVResponse {
+    GenEsiSss(gen_esi_sss::Response),
+    GenPkShareAndSkSss(gen_pk_share_and_sk_sss::Response),
+    GenDecryptionKey(gen_decryption_key::Response),
+    GenDecryptionShare(gen_decryption_share::Response),
+    ThresholdDecrypt(threshold_decrypt::Response),
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub enum TrBFVError {
+    // Add errors here as required
+}
+
+pub mod gen_esi_sss {
```

Comment:
this will include the generation of smudging noise + secret shares from the derived poly, right?

Reply:
This is separate from the pubkey+sk_sss because it can be done in parallel